account_billing/models/account_billing.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import time
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+MAP_INVOICE_TYPE_PARTNER_TYPE = {
+    'out_invoice': 'customer',
+    'in_invoice': 'supplier',
+}
+
+
+class AccountBilling(models.Model):
+    _name = 'account.billing'
+    _description = 'Account Billing'
+    _inherit = ['mail.thread']
+    _order = 'date desc, id desc'
+
+    number = fields.Char(
+        readonly=True,
+        copy=False,
+        help='Number of account.billing',
+    )
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Partner',
+        required=True,
+        default=lambda self: self._get_partner_id(),
+        help='Partner Information',
+        track_visibility='always'
+    )
+    date = fields.Date(
+        string='Billing Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: time.strftime('%Y-%m-%d'),
+        help='Effective date for accounting entries',
+        track_visibility='always'
+    )
+    due_date = fields.Date(
+        string='Due Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: fields.Date.context_today(self),
+        required=True,
+        track_visibility='always'
+    )
+    invoice_ids = fields.One2many(
+        'account.invoice',
+        'billing_id',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self._context.get('active_ids', []),
+        string='Invoices',
+    )
+    invoice_related_count = fields.Integer(
+        string='# of Invoice',
+        compute='_compute_invoice_related_count',
+        help='Count invoice in billing'
+    )
+    state = fields.Selection(
+        [('draft', 'Draft'),
+         ('cancel', 'Cancelled'),
+         ('billed', 'Billed')],
+        string='Status',
+        readonly=True,
+        default='draft',
+        help="""* The 'Draft' status is used when a user is encoding a new and
+            unconfirmed billing.
+            \n* The 'Billed' status is used when user create billing,
+            a billing number is generated
+            \n* The 'Cancelled' status is used when user cancel billing.""",
+    )
+    narration = fields.Text(
+        string='Notes',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Notes"
+    )
+    type = fields.Selection(
```

Comment:
it's better if you don't override this reserved word from python, can you change the name of the variable?

Reply:
@Saran440 how about changing to **bill_type**

Reply:
@Saran440 how about bill_type ?

Reply:
bill_type is okay.